modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java

```diff
@@ -241,10 +236,7 @@ public GraalJSInterceptor getGraalJSInterceptor() {
     }
 
     private static String transformTypeScriptHandlerPathIfNecessary(String handlerPath) {
-        if (handlerPath.endsWith(TS_EXT)) {
-            return handlerPath.substring(0, handlerPath.length() - TS_EXT.length()) + MJS_EXT;
-        }
-        return handlerPath;
+        return handlerPath.endsWith(TS_EXT) ? handlerPath.replaceAll(Pattern.quote(TS_EXT), JS_EXT) : handlerPath;
```

Comment:
This is a breaking change in some cases. Why are we changing the extension to “.js”?

Reply:
With the latest commit the file extension is still changed from `*.ts` to `*.js`, however in the case of `*.ts` the **ModuleType** is assigned to `ModuleType.ESM`